### PR TITLE
Refactor 007, 008 and 009 test scripts

### PR DESCRIPTION
**Pull request by electrocucaracha**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind bug

/kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This change extracts the instructions to get the newest pod name and refactor some other instructions to facilitate the reading of the testing code.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by johnbelamaric** [comment]
/retest

It's weird, the `approve` returned status 1 but there was no output.

**Comment by electrocucaracha** [comment]
> /retest
> 
> It's weird, the `approve` returned status 1 but there was no output.

I'm doing more refactoring on this, but it's not passing the 009.sh test. I'm not sure if we still require [this process](https://github.com/electrocucaracha/test-infra/blob/main/e2e/tests/009.sh#L68-L120) and if that's the case, why don't use it in the 008.sh?

**Comment by johnbelamaric** [comment]
> I'm doing more refactoring on this, but it's not passing the 009.sh test. I'm not sure if we still require [this process](https://github.com/electrocucaracha/test-infra/blob/main/e2e/tests/009.sh#L68-L120) and if that's the case, why don't use it in the 008.sh?

Yes, I think it should be there for now at least. We could also have it in 008.sh, it just didn't flake in the same way. Still not clear why something is going in and tweaking the packages and causing the issue.

**Comment by electrocucaracha** [comment]
@vjayaramrh @johnbelamaric I think this change is ready to be merged.

File: e2e/tests/009.sh [hunk 1]
```diff
@@ -88,14 +82,14 @@ while [[ $retries -gt 0 ]]; do
 
     modified=false
     info "Pushing update"
-    output=$(kpt alpha rpkg push -n default "$smf_pkg_rev" $ws >/dev/null 2>&1)
+    output=$(kpt alpha rpkg push -n default "$smf_pkg_rev" $ws 2>&1)
```
Comment (electrocucaracha) [inline e2e/tests/009.sh:85]: @johnbelamaric @vjayaramrh changes on lines 85, 92 and 102 will fix some CI issues, I think the previous change was sending stdout and stderr to /dev/null

**Pull request merged**